src/functions/findFirst.js

```diff
@@ -136,6 +147,22 @@ export const findFirst = wrap({
             const result = hi("etetet").findFirst("etet");
             hi.assert(result.index === 0);
             hi.assert(result.high === 4);
+        },
+        "emptySearchString": (hi) => {
+            hi.assertFailWith(NotBoundedError, () => {
+                return hi("test").findFirst("");
+            });
+        },
+        "nonStringSearchSequence": (hi) => {
+            hi.assertFailWith(NotBoundedError, () => {
+                return hi(true, false, {}).findFirst({});
+            });
+            hi.assertUndefined(NotBoundedError, () => {
+                return hi(true, false, {}).findFirst("true");
```

Comment:
If the `hi` call actually produced a sequence this would be legal (currently it should be returning `undefined`)